shell/testing/observatory/service_client.dart

```diff
@@ -44,55 +47,46 @@ class ServiceClient {
   }
 
   void _onData(dynamic message) {
-    final Map<String, dynamic> response =
-        json.decode(message as String) as Map<String, dynamic>;
+    final response = json.decode(message as String) as Map<String, Object?>;
     final dynamic key = response['id'];
     if (key != null) {
       print('<- $key');
-      final dynamic completer = _outstandingRequests.remove(key);
-      assert(completer != null);
-      final dynamic result = response['result'];
-      final dynamic error = response['error'];
+      final completer = _outstandingRequests.remove(key)!;
+      final result = response['result'];
+      final error = response['error'];
```

Comment:
Should dynamic be explicit?  I guess we don't have a lint, but it is explicit above.

Reply:
It's `Object?`? Sounds good.